src/components/StudentsTable.vue

```diff
@@ -60,15 +94,70 @@ const onCellEditingStopped = (params: CellEditingStoppedEvent) => {
   if (someValueChanged.value) canSubmit.value = true
 }
 
+const onSelectionChanged = () => {
+  selectedRows.value = studentsTableApi.value?.getSelectedRows() ?? []
+}
+
+const getRowId = (params: GetRowIdParams) => params.data.id.value
+
+const ensureRowsDataIsValid = (rows: StudentWithMetadata[]) =>
+  rows.every((student) => Object.values(student).every(({ isValidated }) => isValidated))
+
 const updateStudents = () => {
   studentsStore.updateStudents({
+    newStudents: addedRows.value,
     onError: (errorMessage: string) => alert(errorMessage),
     onSuccess: () => {
       canSubmit.value = false
       someValueChanged.value = false
+      addedRows.value = []
     }
   })
 }
+
+const addRow = () => {
+  const emptyStudent = {
+    id: { value: crypto.randomUUID(), isValidated: true },
+    name: { isValidated: false },
+    lastName: { isValidated: false },
+    birthDate: { isValidated: false },
+    finalGrade: { isValidated: false },
+    hobbies: { value: [], isValidated: true }
+  }
+
+  studentsTableApi.value?.applyTransaction({
+    add: [emptyStudent]
+  })
+
+  addedRows.value.push(emptyStudent)
+}
+
+const getSelectedStudents = (persisted: boolean) => {
+  return selectedRows.value.filter((selectedStudents) => {
+    const result = addedRows.value.find((addedStudent) => addedStudent.id === selectedStudents.id)
+
+    return persisted ? !result : result
+  })
+}
+
+const deleteSelectedRows = () => {
+  studentsStore.deleteStudents({
+    persistedStudentsToDelete: getSelectedStudents(true),
+    onSuccess: () => {
+      studentsTableApi.value?.applyTransaction({ remove: getSelectedStudents(false) })
+
+      addedRows.value = addedRows.value.filter(
+        (addedStudent) =>
+          !selectedRows.value.find(
+            (selectedStudent) => selectedStudent.id.value === addedStudent.id.value
+          )
+      )
```

Comment:
This implementation is ok, but we have 3 sources of truth for this table:
- store
- ag-grid internals (applyTransaction)
- addedRows

This can lead to problems with keeping that in sync if we plan to scale the app and add more functionality. What do you guys think about having 1 source of truth (store)? We can add a new empty row to students array with some flag indicating that this is not saved to the backend. Ofc I'm not pushing you to change it, just wanted to hear your opinion.

Reply:
Yes, I agree, that was in fact the first approach I tried. However, didn't I need to use some kind of transaction for the grid? When I attempted to add more than one new row with the store as the only source of truth, only the first row was reflected in the grid. The rest were added to the store state but did not appear in the grid.

Do you have any suggestions on how to fix this issue, or I am doing something wrong?